helm-charts/splunk-otel-collector/values.yaml

```diff
@@ -557,8 +563,20 @@ fluentd:
       cpu: 100m
       memory: 200Mi
 
+# To run fluentd container as a non root user and enable other seccurity features,
+# delete runAsUser: 0 and uncomment other securityContext settings
+# NOTE: during internal tests running this container as a non-root user with cri-o runntime didn't work.
```

Comment:
What kind of testing has been done running fluentd as non-root? I'm glad at least some kind of error popped up in testing, anytime something is changed to non-root permissions I'd be pretty concerned about things breaking.

Reply:
We have tested on Minikube with different runtimes (i.e. docker, containerd and cri-o). Also, tested it on [rosa](https://aws.amazon.com/rosa/).

We have followed these steps to test running as a non-root user on each runtime:
1.  Deploy. the collector
2. Deploy datagen pod
3. Verify datagen logs are collected in splunk

Since we have changed the default checkpoint path for fluentd, we have tested the following scenarios:
- fluentd(old version) -> fluentd(current branch)
- fluentd(current branch) -> otel(current branch)
- fluentd(old version) -> otel(current branch)

For the above scenarios, we have validated that it reads the correct checkpoints and continues reading the data from where it left